Oneirocom/Magick
# Agent runner

## What Changed:

- Implemented event streaming from agent/agent server, with subscriptions on server
- implemented new emitter plugin to emit all work done by spells on agent server to main server
- created proper channel based permissions and authentication of realtime messages
- added new agent service events to relay events up to connected clients
- added bullMQ queue to pass work from main server to running agent
- added bullMQ queue to send results of running back from agent server to main server
- added publishing of results from main server to connected clients
- added agent select dropdown to VRM screen for selecting a live running agent to talk to
- hooked up VRM chat to send messages to live running agent, and to subscribe to responses from the agent
- broke out some functionality to stand-alone agents package for logic related to agent server
- extended agent from redis pubsub to give agent pubsub capabilities over redis

## How to test:

- ensure all agent functionality works as expected and is not broken (deployed, runs, functions)
- select an agent from the VRM dropdown and have a chat with it.  VRM should respond as the agent.

## Additional information:

This is a lot of things under the hood for major architecture, so good to ensure that core functionality is all still in place.
Future PR will show the events streaming to the client and we can test some security measures then (users only receiving their own events, etc)

Implements a `LOCAL_DEV` flag which we probably want to set to false in our deploy environments.
